x-pack/packages/kbn-elastic-assistant/impl/assistant/prompt_editor/system_prompt/system_prompt_settings_management/use_system_prompt_table.tsx

```diff
@@ -0,0 +1,131 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { EuiBasicTableColumn, EuiIcon, EuiLink } from '@elastic/eui';
+import React from 'react';
+import { Conversation } from '../../../../assistant_context/types';
+import { AIConnector } from '../../../../connectorland/connector_selector';
+import { BadgesColumn } from '../../../common/components/assistant_settings_management/badges';
+import { RowActions } from '../../../common/components/assistant_settings_management/row_actions';
+import { Prompt } from '../../../types';
+import {
+  getConversationApiConfig,
+  getInitialDefaultSystemPrompt,
+} from '../../../use_conversation/helpers';
+import { SYSTEM_PROMPT_DEFAULT_NEW_CONVERSATION } from '../system_prompt_modal/translations';
+import * as i18n from './translations';
+import { getSelectedConversations } from './utils';
+
+type ConversationsWithSystemPrompt = Record<
+  string,
+  Conversation & { systemPrompt: Prompt | undefined }
+>;
+
+type SystemPromptTableItem = Prompt & { defaultConversations: string[] };
+
+export const useSystemPromptTable = () => {
+  const getColumns = ({
+    onEditActionClicked,
+    onDeleteActionClicked,
+  }: {
+    onEditActionClicked: (prompt: SystemPromptTableItem) => void;
+    onDeleteActionClicked: (prompt: SystemPromptTableItem) => void;
+  }): Array<EuiBasicTableColumn<SystemPromptTableItem>> => [
+    {
+      align: 'left',
+      name: i18n.SYSTEM_PROMPTS_TABLE_COLUMN_NAME,
+      truncateText: { lines: 3 },
+      render: (prompt: SystemPromptTableItem) =>
+        prompt?.name ? (
+          <EuiLink onClick={() => onEditActionClicked(prompt)}>
+            {prompt?.name}
+            {prompt.isNewConversationDefault && (
+              <EuiIcon
+                type="starFilled"
+                aria-label={SYSTEM_PROMPT_DEFAULT_NEW_CONVERSATION}
+                className="eui-alignTop"
+              />
+            )}
+          </EuiLink>
+        ) : null,
+    },
+    {
+      align: 'left',
+      name: i18n.SYSTEM_PROMPTS_TABLE_COLUMN_DEFAULT_CONVERSATIONS,
+      render: ({ defaultConversations, id }: SystemPromptTableItem) => (
+        <BadgesColumn items={defaultConversations} prefix={id} />
+      ),
+    },
+    /* TODO: enable when createdAt is added
+      {
+        align: 'left',
+        field: 'createdAt',
+        name: i18n.SYSTEM_PROMPTS_TABLE_COLUMN_CREATED_ON,
+      },
+      */
+    {
+      align: 'center',
+      name: 'Actions',
+      width: '120px',
+      render: (prompt: SystemPromptTableItem) => {
+        const isDeletable = !prompt.isDefault;
+        return (
+          <RowActions<SystemPromptTableItem>
+            rowItem={prompt}
+            onEdit={onEditActionClicked}
+            onDelete={isDeletable ? onDeleteActionClicked : undefined}
+            isDeletable={isDeletable}
+          />
```

Comment:
As with the other comment on actions, should we just show them since there are only ever two  instead of adding another click into the popover?